src/components/PopupWithForm.js

```diff
@@ -14,26 +14,36 @@ class PopupWithForm extends Popup {
     inputs.forEach((input) => {
       values[input.name] = input.value;
     });
+
     return values;
   }
 
   _submitForm = () => {
     const inputValues = this._getInputValues();
+    this._loadingState = true;
+    this.showLoading(true);
     this._handleFormSubmit(inputValues);
   };
 
+  showLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
+    }
+  }
+
+  hideLoading() {
+    this.showLoading(false);
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
     });
   }
-
-  close() {
```

Comment:
Needs correction: There is no need to remove this, you should reset the form when close the popup